src/components/GoalsComponents/MyGoal/MyGoal.tsx

```diff
@@ -23,6 +24,9 @@ interface MyGoalProps {
 const MyGoal: React.FC<MyGoalProps> = ({ goal, dragAttributes, dragListeners }) => {
   const [expandGoalId, setExpandGoalId] = useState("root");
   const [isAnimating, setIsAnimating] = useState(true);
+  const { copyCode } = useGoalActions();
+  const { title } = goal;
+  const isCode = isGoalCode(title);
```

Comment:
You don't need to define it here since it's only used in one place. Put it directly in the if condition instead.